engine/servermaster/executor_manager.go

```diff
@@ -43,6 +44,7 @@ type ExecutorManager interface {
 	ListExecutors() []string
 	CapacityProvider() scheduler.CapacityProvider
 	GetAddr(executorID model.ExecutorID) (string, bool)
+	WatchExecutors(ctx context.Context) ([]model.ExecutorID, *notifier.Receiver[model.ExecutorID], error)
```

Comment:
Could you explain what's the difference of the first param and second param?

Reply:
a snapshot and a continuous event sender on that snapshot. Also please add some comments.

Reply:
`a continuous event sender on that snapshot`:
I notice we can get the `ExecutorID ` from this receiver, but still can't get what we can do if we can get a  continuous `ExecutorID` event.

Reply:
> `a continuous event sender on that snapshot`: I notice we can get the `ExecutorID ` from this receiver, but still can't get what we can do if we can get a continuous `ExecutorID` event. @lance6716

The stream of ExecutorIDs is meant to be offline events. Sorry for not making it clearer and I've made some changes in the latest commit.

Reply:
Looking forward to your inplementation.
May need to take good care of the dynamic feature of `watching + snapshot` behavior.